packages/@aws-cdk/aws-appmesh/lib/service-discovery.ts

```diff
@@ -45,22 +24,43 @@ export interface ServiceDiscoveryConfig {
   readonly cloudmap?: CfnVirtualNode.AwsCloudMapServiceDiscoveryProperty;
 }
 
+/**
+ * Enum of DNS service discovery response type
+ */
+export enum ResponseType {
+  /**
+   * DNS resolver returns a loadbalanced set of endpoints.
+   */
+  LOADBALANCER = 'LOADBALANCER',
+
+  /**
+   * DNS resolver is returning all the endpoints.
+   */
+  ENDPOINTS = 'ENDPOINTS',
+}
+
 /**
  * Provides the Service Discovery method a VirtualNode uses
  */
 export abstract class ServiceDiscovery {
   /**
    * Returns DNS based service discovery
    */
-  public static dns(hostname: string): ServiceDiscovery {
-    return new DnsServiceDiscovery(hostname);
+  public static dns(hostname: string, responseType?: ResponseType): ServiceDiscovery {
+    return new DnsServiceDiscovery(hostname, responseType);
   }
 
   /**
    * Returns Cloud Map based service discovery
+   *
+   * @param service The AWS Cloud Map Service to use for service discovery
+   * @param instanceAttributes A string map that contains attributes with values that you can use to
+   *                           filter instances by any custom attribute that you specified when you
+   *                           registered the instance. Only instances that match all of the specified
+   *                           key/value pairs will be returned.
```

Comment:
We don't align in docs like this:

```suggestion
   * @param instanceAttributes A string map that contains attributes with values that you can use to
   *   filter instances by any custom attribute that you specified when you
   *   registered the instance. Only instances that match all of the specified
   *   key/value pairs will be returned.
```

Reply:
Well noted. Thanks for the pointer.